core/src/main/java/edu/ucr/cs/riple/core/global/GlobalAnalyzerImpl.java

```diff
@@ -0,0 +1,217 @@
+    // Collect callers of public APIs in module.
+    MethodRegionTracker tracker = new MethodRegionTracker(config.downstreamInfo, tree);
+    // Generate fixes corresponding methods.
+    ImmutableSet<Fix> fixes =
+        methods.values().stream()
+            .filter(
+                input ->
+                    !tracker
+                        .getCallersOfMethod(input.node.location.clazz, input.node.location.method)
+                        .isEmpty()) // skip methods that are not called anywhere.
+            .map(
+                methodImpact ->
+                    new Fix(
+                        new AddAnnotation(
+                            new OnMethod(
+                                "null",
+                                methodImpact.node.location.clazz,
+                                methodImpact.node.location.method),
+                            config.nullableAnnot),
+                        "null",
+                        "null",
+                        "null",
+                        false))
+            .collect(ImmutableSet.toImmutableSet());
+    DownstreamImpactAnalyzer analyzer =
+        new DownstreamImpactAnalyzer(
+            fixes, new DownstreamDependencySupplier(config, tree), tracker);
+    ImmutableSet<Report> reports = analyzer.explore();
+    // Update method status based on the results.
+    methods
+        .values()
+        .forEach(
+            method -> {
+              MethodNode node = method.node;
+              Set<OnParameter> impactedParameters = analyzer.getImpactedParameters(node.location);
+              reports.stream()
+                  .filter(input -> input.root.toMethod().equals(node.location))
+                  .findAny()
+                  .ifPresent(report -> method.setStatus(report, impactedParameters));
+            });
+    System.out.println("Analyzing downstream dependencies completed!");
+  }
+
+  /**
+   * Retrieves the corresponding {@link MethodImpact} to a fix.
+   *
+   * @param fix Target fix.
+   * @return Corresponding {@link MethodImpact}, null if not located.
+   */
+  @Nullable
+  private MethodImpact fetchStatus(Fix fix) {
+    if (!fix.isOnMethod()) {
+      return null;
+    }
+    OnMethod onMethod = fix.toMethod();
+    int predictedHash = MethodImpact.hash(onMethod.method, onMethod.clazz);
+    Optional<MethodImpact> optional =
+        this.methods.get(predictedHash).stream()
+            .filter(m -> m.node.location.equals(onMethod))
+            .findAny();
+    return optional.orElse(null);
+  }
+
+  /**
+   * Returns the effect of applying a fix on the target on downstream dependencies.
+   *
+   * @param fix Fix targeting an element in target.
+   * @param fixesLocation Location in target that will be annotated as {@code @Nullable}.
+   * @return Effect on downstream dependencies.
+   */
+  private int effectOnDownstreamDependencies(Fix fix, Set<Location> fixesLocation) {
+    MethodImpact status = fetchStatus(fix);
```

Comment:
Why status and not just: `MethodImpact methodImpact = fetchMethodImpactForFix(...)`? (renaming the `fetchStatus` method too)

Reply:
Left over from old names/design, thanks for noticing this. 0b5e623bb586ed8874ed90cbcaefdbefd2fe43e7